sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1207,6 +1207,13 @@ object SQLConf {
       .intConf
       .createWithDefault(100)
 
+  val RANGE_EXCHANGE_SAMPLE_CACHE_ENABLE =
+    buildConf("spark.sql.execution.rangeExchange.sampleCache.enabled")
```

Comment:
I think this is a feature of Spark core instead of Spark SQL. `RangePartitioner` is in Spark core and we can apply this optimization to `RDD.sort` as well

Reply:
Yes. The RangePartitioner is mainly used for sorting. Not only in Spark Sql, it is used in many other places. There is no need for config here. I removed the config, and the optimization will work by default.